Creation of ContactMe Section

samuelouteda opened the pull request:
Assembled ContactMe.jsx component: 
![image](https://user-images.githubusercontent.com/74317636/101993466-65c0f100-3c91-11eb-8e6b-2992d094b712.png)

SectionTitle component:
![image](https://user-images.githubusercontent.com/74317636/101993441-314d3500-3c91-11eb-901e-ee45f97638d5.png)

SectionSubtitle component:
![image](https://user-images.githubusercontent.com/74317636/101993448-41651480-3c91-11eb-857b-4acb23ad8a84.png)

InputTextField component:
![image](https://user-images.githubusercontent.com/74317636/101993454-4f1a9a00-3c91-11eb-98f1-7d6cad444dcd.png)

ContactInfo component:
![image](https://user-images.githubusercontent.com/74317636/101993474-77a29400-3c91-11eb-969e-3f99a36f7a93.png)

AssembledForm component:
![image](https://user-images.githubusercontent.com/74317636/101993501-91dc7200-3c91-11eb-81cc-2c5e6bbfd653.png)

Possible FontFamilies for SectionTitle:
![FontFamilies](https://user-images.githubusercontent.com/74317636/101993529-c6502e00-3c91-11eb-821f-43ecb31702de.png)

FontFamilies in Pangrams:
![FontFamilies Pangrams](https://user-images.githubusercontent.com/74317636/101993531-cfd99600-3c91-11eb-8997-12ff12be5f68.png)

SectionTitle in Merriweather's FontFamily:
![image](https://user-images.githubusercontent.com/74317636/102117270-806fa300-3e14-11eb-836a-48efdb0be73e.png)

SectionTitle in Merriweather's FontFamily (Uppercase):
![image](https://user-images.githubusercontent.com/74317636/102117206-6d5cd300-3e14-11eb-8702-b8fd2bc18a1a.png)

samuelouteda committed:
Creation of ContactMe Section

Creation of ContactMe Section assembling three global components and two local components: InputTextField, which styles a Material UI's textfield; SectionTitle, which displays the title of the section with an icon; SectionSubtitle, which displays a given subtitle when needed; ContactInfo, which renders contact information in three aligned blocks, with an icon; and AssembledForm, which imports InputTextField and renders three instances of it, with a styled submit button. More changes were made to other components, like: modifying App.js to render ContactMe.jsx, adding logic to IconMaker to render new icons and variate its sizes, importing the new icons on IconMaker/index.js to be used on IconMaker, adding style rules to style.js, adding icons to ICONS object in constants.js, creating index.js in ContactMe/components to import all the components used in ContactMe.jsx and installing Formik dependencies.

kilordpepo commented on src/components/SectionSubtitle/SectionSubtitle.js:
```diff
@@ -0,0 +1,67 @@
+      <div
+        className={classes.subtitleUnderline}
+        style={{
+          borderColor: borderColor
```
I suppose this can be reduced to ` { borderColor}`, change this here and everywhere else

kilordpepo commented on src/components/SectionTitle/SectionTitle.js:
```diff
@@ -0,0 +1,46 @@
+  }
+}));
+
+const SectionTitle = ({ iconImage, title, family, weight, size }) => {
```
wrap this props as they are styles into an object and then down below just do a spread operator

kilordpepo replied:
if they are different styles for different elemeents then group them into respectives objects

kilordpepo commented on src/sections/ContactMe/components/AssembledForm/AssembledForm.js:
```diff
@@ -0,0 +1,143 @@
+          id='email'
+          name='email'
+          value={formik.values.email}
+          onChange={formik.handleChange}
```
I believe formik has a cleaner way to set this default vaules without you having it to specify it. Please investigate

kilordpepo commented on src/styles/styles.js:
```diff
@@ -59,18 +64,29 @@ export const styles = {
       minWidth: "180px",
       maxWidth: "250px"
+    },
+    contactMeDimensions: {
```
The idea is to make everything more generic so let's not specify per section if not for the whole application. The idea is you use the same spacing or widths in all the application. I would suggest move this to globaldimenssions and use relative values if necessary to avoid creating metrics for every section since we don't have things that different to specify something for a section.

kilordpepo replied:
Please also start using Flex as a display which I believe his props and flow will help you out for this and the rest of the sections: https://css-tricks.com/snippets/css/a-guide-to-flexbox/

samuelouteda committed:
Improving Responsive with flexbox

Improving Responsive Design with flexbox in ContactMe.jsx, modifying ContactMe style rules in styles.js to be more generic, applying these new rules in ContactMe.jsx and ContactInfo, using Property Value Shorthand to assing the received prop borderColor to its corresponding style tag in SectionSubtitle, grouping related style props of SectionTitle in an object and assigning them with a spread operator to its corresponding style tag, adding background to App.js

samuelouteda commented:
I reduced the vertical margins so that ContactMe would fit on the screen without scrolling, do you think it looks good?
![image](https://user-images.githubusercontent.com/74317636/102135988-2cbe8300-3e2f-11eb-9d92-94a119bebef9.png)

samuelouteda committed:
Changing style

Removing Title border from ContactMe.jsx and adding it to SectionTitle, removing unnecesary margins in SectionSubtitle and passing props to edit padding and alignItems

samuelouteda committed:
Creating StyledButton global component

Extracting SubmitButton from AssembledForm and converting it into a new global component, called StyledButton. Importing it on the index.js of ContactMe/components to be exported and used in AssembledForm, instead of SubmitButton.

kilordpepo commented:
Please leave the title without capitals

samuelouteda committed:
Applying Destructuring on AssembledForm

Applying Destructuring on AssembledForm to reduce the use of formik object, adding PropTypes to StyledButton and removing unnecessary makeStyles Hook, adding colors and fontSize rules for subtitles in styles.js and applying the last one on SectionSubtitle
